grep-00XX-relicense-volk.md

```diff
@@ -0,0 +1,86 @@
+This may explicitly include proprietary projects.
+We want to enable all contributors to be able to use VOLK wherever they want.
+At the same time, we want to make sure that improvements to VOLK itself are easily obtained by everyone, i.e. strike a balance between permissiveness and strict copyleft.
+Previously, several attemps were made to relicense VOLK without success.
+With a formal process and a clear vision laid out, we hope to be able to get everyone on board.
+
+Since VOLK is a library it should choose a fitting license.
+If we see greater adoption of VOLK in more project, we hope to also see more contributions.
+May it be bug fixes, new kernels or even contributions to core features.
+
+## Description
+
+In order to facilitate this relicensing effort, we need to got through several steps.
+
+### Notification
+We consider it to be very important to let every contributor know that the wish to relicense by some contributors exists.
+A first step would be to notify every contributor of this GREP.
+
+### Another license discussion
+Currently, the proposed license is LGPLv3+.
+Though, other licenses, possibly more permissive are available too.
+This includes [3-Clause BSD](https://opensource.org/licenses/BSD-3-Clause), [MIT](https://opensource.org/licenses/MIT) and [Apache-2.0](https://www.apache.org/licenses/LICENSE-2.0).
+Prelimenary discussion on the subject matter revealed that Apache-2.0 is out of favor by some.
+We assume that LGPLv3+ strikes a balance between permissiveness and obligation to provide modified source code.
+
+### All contributors need to sign off for the new license
+
+Before we can adopt a new license, we need approval by every contributor to basically resubmit their contributions under the new license.
+[There is more to this discussion and how this evolved but others are more qualified to add to that discussion]
+
+### The process
+We need to figure out how we collect all the contributor approvals.
+This might be a file within the VOLK repo where everyone adds a signed-off commit to approve the new license.
+Currently, we have around 65 contributors to VOLK.
+
+### The meantime
+It might take time to gain approval by every contributor.
+We want to be able to continue VOLK development while we are in this license transition period.
+We need to work out a clear path how to deal with contributions in the mean time.
+This will probably prior include relicensing approval by a majority of contributors.
+Then, we will probably state that new contributions will only be accepted under the terms of the new license.
+The details need to be worked out.
+
+### The aftermath
+Assume most contributors approved to relicense VOLK but some did not.
+How do we deal with that situation?
+A contributor who does not approve the new license, cannot be forced or pushed to do so.
+Either, we decide to give up on this relicensing effort or, we are required to remove all contributions by that contributor.
```

Comment:
Third option, fork off a volk-gpl library that's not accepting new kernels, and is the home of these. Question is whether that isn't a dangerous / work-heavy thing to do.

Reply:
Actually, it is quite cumbersome to maintain several versions. So far, we are quite happy to maintain only one branch `master`. We might want to rename that branch to `main`. Though, that's a different discussion.

Reply:
Yes, I recommend renaming.

And yeah, the other option is undertaking a new implementation effort. There's no way it wouldn't be work intensive.

Reply:
The [Rename default branch issue](https://github.com/gnuradio/volk/issues/461) is open now.

Reply:
@bhilburn are you in favour of renaming master -> main, or renaming VOLK to something else?

Reply:
My comment was re: renaming `master` to `main`

Reply:
I addressed the discussion here and the "renaming" confusion discussion is carried on in a new issue. I mark this discussion as resolved.